### PR TITLE
Downgrade to netcoreapp3.1

### PR DESCRIPTION
**Pull request by JoshLove-msft**
This change would make it easier for the .NET repo to use a build time dependency on the Test Proxy.

Also, netcoreapp3.1 is an LTS version whereas .NET 5.0 is not.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Azure.Sdk.Tools.TestProxy.csproj [hunk 1]
```diff
@@ -11,7 +10,6 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.10.0" />
-    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="5.0.2" />
```
Comment (JoshLove-msft) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Azure.Sdk.Tools.TestProxy.csproj:14]: I don't think this was used.

**Comment by pakrym** [comment]
Do we need to update the docker file or some other infrastructure?

**Comment by JoshLove-msft** [comment]
> Do we need to update the docker file or some other infrastructure?

Good point - I think we would at least need to update:
- https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/docker/dockerfile
- https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/docker/dockerfile-win

though I'm not clear on how to actually test that the change is working.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Azure.Sdk.Tools.TestProxy.csproj [hunk 2]
```diff
@@ -1,7 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFrameworks>net5.0</TargetFrameworks>
-    <RequiredTargetFrameworks>$(TargetFrameworks)</RequiredTargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
```
Comment (weshaggard) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Azure.Sdk.Tools.TestProxy.csproj:3]: Might be worth adding a comment here of why we are keeping it here and only move it if we move the lower .NET test framework.

**Comment by JoshLove-msft** [comment]
> > Do we need to update the docker file or some other infrastructure?
> 
> Good point - I think we would at least need to update:
> 
> * https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/docker/dockerfile
> * https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/docker/dockerfile-win
> 
> though I'm not clear on how to actually test that the change is working.

@scbedd will be checking on this.

**Review by scbedd — COMMENTED** [review]
I'll definitely help make changes on docker, but I'm seeing the following locally.

1. We need to update [ci.yml](https://github.com/Azure/azure-sdk-tools/blob/d94cac8e55e70002d1bb98f7b56858a3efb07463/tools/test-proxy/ci.yml#L28) to target 3.1
2. Update the targeted base docker image versions
3. Update the paths that copy dotnet-devcert out of the SDK -> runtime docker builds.

I'll help with any, but I don't have `write` permissions to the branch.

File: tools/test-proxy/ci.yml [hunk 3]
```diff
@@ -25,4 +25,4 @@ extends:
   template: /eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml
   parameters:
     ToolDirectory: tools/test-proxy
-    DotNetCoreVersion: "5.0.100"
\ No newline at end of file
+    DotNetCoreVersion: "3.1"
```
Comment (JoshLove-msft) [inline tools/test-proxy/ci.yml:28]:
```suggestion
    DotNetCoreVersion: "3.x"
```

**Comment by JoshLove-msft** [comment]
Ci 

> I'll definitely help make changes on docker, but I'm seeing the following locally.
> 
> 1. We need to update [ci.yml](https://github.com/Azure/azure-sdk-tools/blob/d94cac8e55e70002d1bb98f7b56858a3efb07463/tools/test-proxy/ci.yml#L28) to target 3.1
> 2. Update the targeted base docker image versions
> 3. Update the paths that copy dotnet-devcert out of the SDK -> runtime docker builds.
> 
> I'll help with any, but I don't have `write` permissions to the branch.

The CI is now passing. I think we still need to do #2 and #3.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Azure.Sdk.Tools.TestProxy.csproj [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
-    <RequiredTargetFrameworks>$(TargetFrameworks)</RequiredTargetFrameworks>
+    <!--Use netcoreapp3.1 as it is the oldest LTS version that the .NET repo targets-->
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
     <NoWarn>$(NoWarn);1591;AZC0001;AZC0012;CA1724;CA1801;CA1812;CA1822;SA1028</NoWarn>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <IsPackable>true</IsPackable>
     <PackAsTool>true</PackAsTool>
     <ToolCommandName>test-proxy</ToolCommandName>
-    <LangVersion>9.0</LangVersion>
+    <LangVersion>preview</LangVersion>
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Azure.Sdk.Tools.TestProxy.csproj:10]: ❤️  that's an awesome workaround. Didn't know this would work :)

**Comment by scbedd** [comment]
Just a heads up I'm actively working on these docker issues. Hoping I'll have a fix sometime tomorrow.

File: tools/test-proxy/docker/dockerfile-win [hunk 5]
```diff
@@ -37,11 +41,12 @@ USER ContainerAdministrator
 RUN mkdir certwork & mkdir srv & cd srv & mkdir testproxy
 ADD docker_build/dotnet-devcert.pfx certwork
```
Comment (scbedd) [inline tools/test-proxy/docker/dockerfile-win:43]:
```suggestion
# we still need to import this certificate to ensure it's available to all users. there is _something_ gnarly going on that purely having the cert 
# available through the environment variables is causing some inconsistent weirdness.    
```

**Review by scbedd — APPROVED** [review]

**Comment by scbedd** [comment]
@JoshLove-msft confirmed working. It's not the best solution, as it does expand our files somewhat, but I need to unblock you and `powershell` trusting isn't a silver bullet either. 

We'd need to do a linux version of the importing that `dotnet dev certs` is doing on linux, and given that the `powershell` one isn't even working properly for me on windows, I don't want to delay your PR by some "I don't know until it's solved" measure.

**Comment by JoshLove-msft** [comment]
> @JoshLove-msft confirmed working. It's not the best solution, as it does expand our docker images somewhat, but I need to unblock you and `powershell` trusting isn't a silver bullet either.
> 
> We'd need to do a linux version of the importing that `dotnet dev certs` is doing on linux, and given that the `powershell` one isn't even working properly for me on windows, I don't want to delay your PR by some "I don't know until it's solved" measure.

For now we still have the dependency on 5.0 in the .NET repo, so I don't think it is urgent that we merge this.

**Comment by pakrym** [comment]
>For now we still have the dependency on 5.0 in the .NET repo, so I don't think it is urgent that we merge this.

The test-proxy is the last thing that requires it.

**Comment by JoshLove-msft** [comment]
> > For now we still have the dependency on 5.0 in the .NET repo, so I don't think it is urgent that we merge this.
> 
> The test-proxy is the last thing that requires it.

Yes, but if this would break dev certs for linux, I don't think we should merge it yet.

**Comment by JoshLove-msft** [comment]
> > > For now we still have the dependency on 5.0 in the .NET repo, so I don't think it is urgent that we merge this.
> > 
> > 
> > The test-proxy is the last thing that requires it.
> 
> Yes, but if this would break dev certs for linux, I don't think we should merge it yet.

Or is everything currently working right now with the latest changes @scbedd?

**Comment by scbedd** [comment]
@JoshLove-msft it's not an optimal solution but it's definitely working 100%. Confirmed by hitting up `https://localhost:5001/Info/Available` on my local box on both versions. We should be good with this merge.

**Pull request merged**